### PR TITLE
114a multiple savegames redo

### LINKED ISSUE #114: Multiple save games
Currently, the save game will always be written to `sav/save.json`. This should be changed to support multiple save games, one for each player.

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #114 

This time it actually works!
Popups and SaveGames are refactored.
Also, you can now choose a file to be loaded

<feature, review, organisation

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/gui/MainMenuScene.java [hunk 1]
```diff
@@ -85,39 +90,90 @@
...
+		return !(name.contains(".") | name.contains("/") | name.contains("\\"));
+	}
+
+	/**
+	 * Check whether the given input corresponds to
+	 * an already existing savefile. If so, load
+	 * that file, otherwise create a new game with that name.
+	 */
+	private static void createNewGame() {
+		if(State.getPlayerName() != null) {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/MainMenuScene.java]: *Auto-generated comment*

    'if' is not followed by whitespace.

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java [hunk 2]
```diff
@@ -244,22 +257,25 @@ private void pausePopup(final int buttonNo) {
 
 	private void playerDies() {
 		State.getTrack().getPlayer().die();
-		sleep(2 * InternalTicker.NANOS_PER_TICK / InternalTicker.E6);
+		sleep(20 * InternalTicker.NANOS_PER_TICK / InternalTicker.E6);
```
Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: 20 is a magic number. Please check your tests with SAT.

<review,configuration

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java [hunk 3]
```diff
@@ -244,22 +257,25 @@ private void pausePopup(final int buttonNo) {
...
 		letPlayerSurvive();			// Make sure there are no obstacles
 		sleep(LONG);
 	}
 
-	private void deathPopup(final int buttonNo) {
-		if (GameScene.getPopup() == null) {
-			fail("The Death Popup is not available.");
+	private void clickPopup(final int buttonNo) {
+		final AbstractScene scene = ShaftEscape.getScene();
+		if (scene.getPopup() != null) {
```
Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: scene.getPopup() should be accessed in a static way.

At least, that's what Eclipse says. And I think the same.

<review,logic

**Comment by mpsijm** [comment]
When I enter my name, play a game, die, return to main menu, press load game, my name isn't in the list.

<bug, review

**Comment by MathiasMeuleman** [comment]
Thank you @mpsijm 
I fixed the things you asked for.

<review

File: src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java [hunk 4]
```diff
@@ -47,7 +46,8 @@
 	 */
 	@Override
 	public final void start(final Stage appStage) {
-		State.resetAll();
+		State.
+				resetAll();
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/ShaftEscape.java]: What happened here?

< layout

File: src/main/java/nl/tudelft/ti2206/group9/gui/MainMenuScene.java [hunk 5]
```diff
@@ -85,39 +90,90 @@
...
+	 *     <li>Empty name</li>
+	 *     <li>Name contains a '.'</li>
+	 *     <li>Name contains a '/'</li>
+	 *     <li>Name contains a '\'</li>
+	 * </a>
+	 * @param name the name to be checked
+	 * @return boolean to indicate whether the name is valid
+	 */
+	private boolean checkPlayerName(final String name) {
+		return !(name.contains(".") | name.contains("/") | name.contains("\\"));
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/gui/MainMenuScene.java]: Findbugs complains here, as said earlier in this PR.

File: src/main/java/nl/tudelft/ti2206/group9/gui/WarningPopup.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import javafx.scene.text.Text;
+import javafx.stage.Popup;
+
+import java.util.Arrays;
+
+/**
+ * @author Mathias
+ */
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/gui/WarningPopup.java]: Can you add a @SuppressWarnings("restriction") here please? :)

<review,organization

**Comment by mdingjan** [comment]
I reviewed your changes and checked them with our SAT. Please have a look at the comments above.
When you've processed all feedback, I think this branch is ready to be merged.

<review,logic

**Comment by thervh70** [comment]
If travis is satisfied with this branch, i am too. I will merge when he says go.

< review

**Comment by mdingjan** [comment]
I see that my suggestions are implemented. Thanks!

@thervh70 You have my permission.

<review

**Pull request merged**